module/zfs/qat_compress.c

```diff
@@ -379,6 +413,7 @@ qat_compress(qat_compress_dir_t dir, char *src, int src_len,
 
 		compressed_sz = dc_results.produced;
 		if (compressed_sz + hdr_sz + ZLIB_FOOT_SZ > dst_len) {
+			*c_len = src_len;
```

Comment:
This line is useless, as if QAT returns fail, software will take over and do it again.